Sources/Plasma/PubUtilLib/plPipeline/GL/plGLPipeline.cpp

```diff
@@ -553,6 +553,7 @@ void plGLPipeline::RenderSpans(plDrawableSpans* ice, const hsTArray<int16_t>& vi
                                 material,
                                 tempIce.fVStartIdx, tempIce.fVLength,   // These are used as our accumulated range
                                 tempIce.fIPackedIdx, tempIce.fILength );
+            glDeleteVertexArrays(1, &vao);
```

Comment:
To fix a memory leak I'm deleting the vertex array after it's not needed any more. I was up to a few gigs after about 5 minutes running a more complicated age. Mac performance is still kind of iffy, so I've been looking at why. The double glFlush in the Mac eglCocoa file was definitely contributing.

Not sure if there is a way to cache these vertex arrays for objects that aren't stale. That could probably help improve performance as well. But for now, definitely have to delete them after they're done.

Reply:
Everything started out with just vertex buffers and index buffers, and then my first attempt to get it running on macOS was unhappy that I didn't have vertex arrays. So I did the bare minimum to make it run without complaining.

I think ultimately we do want to use vertex arrays to hopefully speed up rendering (and I'm pretty sure I did get that set up for the plGLPlateManager for stuff like the cursor), but it's not clear exactly where to stick the VAO reference and it's currently unused. :+1: to clean it up for now.